extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -273,6 +290,81 @@ export class AmpAnalytics extends AMP.BaseElement {
       return val + argList;
     });
 
+    if (request.startsWith("iframe")) {
+      this.handleIframe_(request, trigger, unusedEvent);
+    }
+    else {
+      this.handleGet_(request, trigger, unusedEvent);
+    }
+  }
+
+  /**
+   * Callback for events that are registered by the config's iframe triggers.
+   * This method generates the request and sends the request out.
+   *
+   * @param {!JSONObject} request string which contains the url to be loaded
+   *                              into the iframe
+   * @param {!JSONObject} trigger JSON config block that resulted in this event.
+   * @param {!Object} unusedEvent Object with details about the event.
+   * @private
+   */
+  handleIframe_(request, trigger, unusedEvent) {
+    const vars = this.mergeObjects_(this.config_.vars, trigger.vars);
+    vars.trigger = trigger;
+    vars.event = unusedEvent;
+
+    // While queuing is enabled load all requests into the queue
+    if (this.iframeQueueing_) {
+      this.iframeQueue_.push(vars);
+    }
+
+    // has this iframe been used to analytics requests
+    let iframe = this.loadedIframes_[trigger.request];
+
+    if (!iframe) {
+      // does the iframe exist on the page somewhere else
+      iframe = document.getElementById(trigger.request);
+      if (iframe && iframe.nodeName.toLowerCase() != "iframe") {
```

Comment:
Sure thing! In the original implementation all analytics calls are made by immediate requests. Using the iframe approach a race condition was created where there is a lag between the first analytics call being made and the handling iframe loading. To minimize this window I made a queue to handle all requests until the iframe is loaded. Once it is loaded the queuing is turned off and the queue is emptied.